props/saber_fett263_buttons.h

```diff
@@ -293,39 +659,745 @@ class SaberFett263Buttons : public virtual PropBase {
+        sound_queue_.Play(SoundToPlay("mcolorsb.wav"));
+        break;
+      case MENU_COLOR_MODE:
+        sound_queue_.Play(SoundToPlay("mcolorop.wav"));
+        break;
+#if (NUM_BLADES > 1)
+      case MENU_BLADE_COLOR:
+        sound_queue_.Play(SoundToPlay("mblade.wav"));
+        break;
+      case MENU_BLADE_STYLE:
+        sound_queue_.Play(SoundToPlay("mblade.wav"));
+        break;
+      case MENU_BLADE_LENGTH:
+        sound_queue_.Play(SoundToPlay("mblade.wav"));
+        break;
+#endif
+      case MENU_GESTURE_SUB:
+        sound_queue_.Play(SoundToPlay("mgestsub.wav"));
+        break;
+      case MENU_SETTING_SUB:
+        sound_queue_.Play(SoundToPlay("msetsub.wav"));
+        break;
+      case MENU_EFFECT:
+        sound_queue_.Play(SoundToPlay("meffect.wav"));
+        break;
+      case MENU_STYLE_SETTING_SUB:
+        sound_queue_.Play(SoundToPlay("mstylstm.wav"));
+        break;
+    }    
+  }
+
+  void SetCopy() {
+    effect_num_ = 16;
+    char set_copy[16];
+    while (true) {
+      effect_num_ -= 1;
+    if (style_parser.UsesArgument(current_preset_.GetStyle(blade_num_), effect_num_ + 2)) break;
+    }
+    saved_color_ = GetColorArg(blade_num_, effect_num_);
+    hsl_ = saved_color_.toHSL();
+    NewColor(blade_num_, effect_num_);
+    current_preset_.Save();
+    effect_num_ = 0;
+  }
+
+  void SetInOut() {
+    char style_arg[10];
+    itoa(calc_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    current_preset_.Save();
+    switch (menu_type_) {
+      case MENU_STYLE_OPTION:
+      case MENU_IGNITION_TIME:
+      case MENU_IGNITION_OPTION:
+      case MENU_PREON_OPTION:
+      case MENU_PREON_SIZE:
+        UpdateFont(current_preset_.preset_num);
+        break;
+      case MENU_RETRACTION_TIME:
+      case MENU_RETRACTION_OPTION:
+        UpdateStyle(current_preset_.preset_num);
+        off_event_ = true;
+        last_rotate_ = millis();
+        break;
+      default:
+        break;
+    }
+  }
+
+  void RevertInOut() {
+    char style_arg[10];
+    itoa(arg_revert_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    switch (menu_type_) {
+      case MENU_RETRACTION_OPTION:
+      case MENU_RETRACTION_TIME:
+        char ig[10];
+        itoa(ignite_time_, ig, 10);
+        current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 19, ig));
+        break;
+      default:
+        break;
+    }
+    current_preset_.Save();
+    UpdateFont(current_preset_.preset_num);
+  }
+
+  void SaveStyleSetting() {
+    char style_arg[10];
+    itoa(calc_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    current_preset_.Save();
+    switch (set_num_) {
+      case 21:
```

Comment:
I think we should define an enum for these values.
I'm not entirely sure what set_num_ is (no comment) and I'm not entirely sure what 21/22/23 refers to.

Reply:
Once the num ArgumentNames enum is in, we can use that, right?

Reply:
enum values added in last update